text/0002-sarif-result-schema.md

```diff
@@ -0,0 +1,1939 @@
+      * locale.
+      */
+      fullName?: string;
+
+      /**
+      * A unique identifer for the tool component in the form of a GUID.
+      */
+      guid?: string;
+
+      /**
+      * The minimum value of localizedDataSemanticVersion required in translations consumed by this component; used by
+      * components that consume translations.
+      */
+      minimumRequiredLocalizedDataSemanticVersion?: string;
+
+      /**
+      * The name of the tool component.
+      */
+      name: string;
+
+      /**
+      * An array of reportingDescriptor objects relevant to the notifications related to the configuration and runtime
+      * execution of the tool component.
+      */
+      notifications?: ReportingDescriptor[];
+
+      /**
+      * The organization or company that produced the tool component.
+      */
+      organization?: string;
+
+      /**
+      * A product suite to which the tool component belongs.
+      */
+      product?: string;
+
+      /**
+      * A localizable string containing the name of the suite of products to which the tool component belongs.
+      */
+      productSuite?: string;
+
+      /**
+      * A string specifying the UTC date (and optionally, the time) of the component's release.
+      */
+      releaseDateUtc?: string;
+
+      /**
+      * An array of reportingDescriptor objects relevant to the analysis performed by the tool component.
+      */
+      rules?: ReportingDescriptor[];
+
+      /**
+      * The tool component version in the format specified by Semantic Versioning 2.0.
+      */
+      semanticVersion?: string;
+
+      /**
+      * A brief description of the tool component.
+      */
+      shortDescription?: MultiformatMessageString;
+
+      /**
+      * The tool component version, in whatever format the component natively provides.
+      */
+      version?: string;
+
+      /**
+      * Key/value pairs that provide additional information about the tool component.
+      */
+      properties?: PropertyBag;
+  }
+
+  /**
+  * The runtime environment of the analysis tool run.
+  */
+  export interface Invocation {
+      /**
+      * An array of strings, containing in order the command line arguments passed to the tool from the operating
+      * system.
+      *
+      * CHECKUP: args go heree 
+      */
+      arguments?: string[];
+
+      /**
+      * The command line used to invoke the tool.
+      */
+      commandLine?: string;
+
+      /**
+      * The Coordinated Universal Time (UTC) date and time at which the run ended. See "Date/time properties" in the
+      * SARIF spec for the required format.
+      * 
+      * CHECKUP: this property along with startTimeUtc will contain timing info
+      */
+      endTimeUtc?: string;
+
+      /**
+      * The environment variables associated with the analysis tool process, expressed as key/value pairs.
+      * 
+      * CHECKUP: this property will contain the flags, config, configHash 
```

Comment:
Do they explicitly mean env vars only? Are we overloading this with putting flags/config, etc in?

Reply:
Additionally, oclif provides a [boatload of env var details](https://oclif.io/docs/config) when the CLIs are run. I believe they're in the `config` object (`this.config` in a command).

Reply:
They _may_ explicitly mean env vars - the spec wasnt super clean imo. 

```
An invocation object MAY contain a property named environmentVariables whose value is an object. The property names in this object SHALL contain the names of all the environment variables in the tool's execution environment. The value of each property SHALL be a string containing the value of the specified environment variable. If the value of the environment variable is an empty string, the corresponding property value SHALL be an empty string.

NOTE 1: Environment variable names and values are likely to reveal highly sensitive information. For example, on a Windows machine, environment variables reveal the directories on the execution path, user account name, machine name, logon domain controller, etc.

NOTE 2: The result of setting an environment variable to an empty string is operating system-dependent. On Windows, it removes the variable from the environment. In Unix, an environment variable can have an empty value.

```
 

What do you think @scalvert ?

Reply:
I think the 'may' part was referring to the fact that `environmentVariables` may be present or not, not whether it may contain env vars or not. I think we'd be overloading this if we used it.